controllers/etcd_controller.go

```diff
@@ -963,33 +961,53 @@ func (r *EtcdReconciler) updateEtcdErrorStatus(etcd *druidv1alpha1.Etcd, lastErr
 }
 
 func (r *EtcdReconciler) updateEtcdStatus(etcd *druidv1alpha1.Etcd, svc *corev1.Service, ss *appsv1.StatefulSet) error {
+
+	sts, err := r.waitUntilStatefulSetReady(ss)
```

Comment:
I'd not do this in the `updateStatus` function but in the reconciliation function.

Reply:
We should do this at the beginning of reconciliation. Also, then we can avoid the wait and simply `ReconcileAfter` if the `StatefulSet` is not ready.

Reply:
How does it make sense to check the statefulset status at the beginning if we already know that subsequent steps might UPDATE it, potentially fixing a broken state?

Reply:
👍I misspoke about the beginning part. I meant whenever we do the check for statefulset in the overall reconciliation. The reconciling after part stands still.